### PR TITLE
Initialize LB4 document translate for zhCN

### PR DESCRIPTION
**Pull request by mikefoxnobel**
And translated some pages.

### CODE REVIEW

**Comment by mikefoxnobel** [comment]
@dhmlau 
I cleared that bad PR, and recreated this one.
I didn't copy English version to here this time, but still keeping the English version image.

**Comment by dhmlau** [comment]
Thanks @mikefoxnobel. Assigning this to @nabdelgadir and @agnes512 who were reviewing your previous PR.

File: pages/zh/lb4/index.md [hunk 1]
```diff
@@ -9,25 +9,36 @@ permalink: /doc/zh/lb4/index.html
 summary:
```
Comment (agnes512) [inline pages/zh/lb4/index.md]: Let's change `sidebar` to sidebar: zh_lb4_sidebar.

And feel free to add and translate `summary: LoopBack is a platform for building APIs and microservices in Node.js` to `summary` :smile:

File: pages/zh/lb4/sidebars/lb4_sidebar.yml [hunk 2]
```diff
@@ -0,0 +1,665 @@
+## definition of `menuitem`===========
```
Comment (agnes512) [inline pages/zh/lb4/sidebars/lb4_sidebar.yml]: Let's change the file path to `_data/sidebars/zh_lb4_sidebar.yml`

File: pages/zh/lb4/sidebars/lb4_sidebar.yml [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+# title: string
+# url: string
+# output: [ enum{web, pdf} ] (expressed as comma separated string in YAML)
+# children?: [ menuitem ]
+
+## top level property=================
+# entries: [ menuitem ]
+
+title: LoopBack 4
+url: index.html
```
Comment (agnes512) [inline pages/zh/lb4/sidebars/lb4_sidebar.yml]: Please change the `url` to `url: /doc/zh/lb4/index.html`.
And also add `translated: true` as this page is translated.

File: pages/zh/lb4/sidebars/lb4_sidebar.yml [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+children:
+
+- title: '总览'
+  url: index.html
```
Comment (agnes512) [inline pages/zh/lb4/sidebars/lb4_sidebar.yml]: For the rest of entries, please change the url to `url: /doc/zh/lb4/xxx.html` format.
And add `translated: true/false` to denote the page is translated or not.

e.g the `Getting-started.html` would be `translated: true` and `Relation.html` would be `translated: false`.

File: pages/zh/lb4/Application.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
+---
+lang: zh
+title: '应用（Application）'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
Comment (agnes512) [inline pages/zh/lb4/Application.md]: Let's change **ALL** these translated pages' sidebar to `sidebar: zh_lb4_sidebar`.

**Review by agnes512 — COMMENTED** [review]
Thank you for the contribution! Let's make some changes so that these pages can be generated properly on the site.

**Comment by agnes512** [comment]
Please add a `Not-yet-translated.md` page in `pages/zh/lb4/Not-yet-translated.md`:
```
---
title: "Not yet translated"
lang: zh
layout: stub
sidebar: zh_lb4_sidebar
permalink: /doc/zh/lb4/Not-yet-translated.html
summary:
---

For instructions on how to translate, see [Translating articles](../contrib/Translating_articles.html).
```

**Comment by agnes512** [comment]
Please add `zh_lb4_sidebar` to `_includes/custom/sidebarconfigs.html`:
```
{% elsif page.sidebar == "zh_lb4_sidebar" %}
    {% assign sidebar = site.data.sidebars.zh_lb4_sidebar %}
```

And add `zh_lb4_sidebar` to `_config.yml`file.

Then add `title` and `url` to https://github.com/altyjp/loopback.io/blob/98656c1aecc5720c43c9c2234b897e84867735e1/_data/topnav.yml#L68.

**Comment by agnes512** [comment]
> I didn't copy English version to here this time, but still keeping the English version image.

And also, since the `Not yet translated` page has a link to English page [example](https://loopback.io/doc/ja/lb4/Not-yet-translated.html?url=/doc/ja/lb4/Concepts.html&title=Key%20concepts), I don't think we need all these images.
Sorry that I didn't notice it in the previous PR.

File: pages/zh/lb4/sidebars/lb4_sidebar.yml (hunk 2 above)
Comment (mikefoxnobel) [reply pages/zh/lb4/sidebars/lb4_sidebar.yml]: Moved

File: pages/zh/lb4/sidebars/lb4_sidebar.yml (hunk 3 above)
Comment (mikefoxnobel) [reply pages/zh/lb4/sidebars/lb4_sidebar.yml]: Added "translated: true / false"

File: pages/zh/lb4/sidebars/lb4_sidebar.yml (hunk 4 above)
Comment (mikefoxnobel) [reply pages/zh/lb4/sidebars/lb4_sidebar.yml]: Added "translated: true / false"

File: pages/zh/lb4/Application.md (hunk 5 above)
Comment (mikefoxnobel) [reply pages/zh/lb4/Application.md]: Changed all "sidebar"

File: pages/zh/lb4/index.md (hunk 1 above)
Comment (mikefoxnobel) [reply pages/zh/lb4/index.md]: Added

**Comment by mikefoxnobel** [comment]
@dhmlau Do I must solve the check errors for "commits in this PR have formatting problems" and "PR out of date, needs to be rebased"?

File: _data/topnav.yml [hunk 6]
```diff
@@ -48,23 +48,25 @@ topnav_dropdowns:
...
         - title: "Pусский (v2)"
           url: /doc/ru/lb2/
         - title: "中文 (v2)"
           url: /doc/zh/lb2/
-        - title: "Français (v4)"
-          url: /doc/fr/lb4/
-        - title: "日本語 (v4)"
-          url: /doc/ja/lb4/
+        - title: "中文 (v4)"
+
```
Comment (agnes512) [inline _data/topnav.yml]: Please also add `url` here

File: _data/sidebars/zh_lb4_sidebar.yml [hunk 7]
```diff
@@ -0,0 +1,820 @@
...
+# url: /doc/zh/lb4/string
+# output: [ enum{web, pdf} ] (expressed as comma separated string in YAML)
+# children?: [ menuitem ]
+
+## top level property=================
+# entries: [ menuitem ]
+
+title: LoopBack 4
+url: /doc/zh/lb4/index.html
+children:
```
Comment (agnes512) [inline _data/sidebars/zh_lb4_sidebar.yml]: Since you also translated `index.html` here. Let's add `translated: true` here as well

**Review by agnes512 — COMMENTED** [review]
Almost looks good to me. just a few comments.
And after this, please rebase, and squash the commits. Then I think it's ready to ship :smile:

**Comment by mikefoxnobel** [comment]
@agnes512 I did rebase, it seems not solve the problem and make things worse. Please help.

**Comment by agnes512** [comment]
Sure! these are steps to rebase, squash, and edit the commits:
```
git checkout gh-pages
git pull
git checkout YOURBRANCH
git rebase -i gh-pages // rebase and squash commits
``` 
then you will see
Screen Shot 2019-09-24 at 8 22 58 AM

edit every commit _but the first one_ from `pick` to `f` then save it:
Screen Shot 2019-09-24 at 8 23 23 AM
You might need to resolve some conflicts.
If you do, just solve it, save the files, `git add .`, then do `git rebase --continue`. `wq` the  message shows after. Repeat till you rebase successfully. 

The last step is to edit the commit message so that the PR linter passes:
`git commit --amend`
then edit the commit message to `Initialize LB4 document translate for zhCN ` or `add Chinese translation for LB4`. Make sure that the commit message is less then 50 characters, or break it to several lines, e.g:
change
```
my origin commit message is way too long the ci fails what should i do to make it pass this is for loopback translation more information here...
```
to
```
first line: break my commit message into lines
// the second line should be blank
more info in the third line ...
```

Hope this helps!

**Pull request closed without merging**